src/Parsers.jl

```diff
@@ -235,6 +235,28 @@ function parse_header_field(bytes::SubString{String})::Tuple{Header,SubString{St
         return (group(1, re, bytes) => unfold), nextbytes(re, bytes)
     end
 
+    # https://github.com/JuliaWeb/HTTP.jl/issues/796
+    # there may be legacy values encoded in latin-1 that will fail
+    # the utf8-expecting regexes above, before throwing the error here,
+    # we can do the ascii check and potentially re-encode the headers and
+    # try again
+    rawbytes= codeunits(bytes)
+    n = count(≥(0x80), rawbytes)
+    if n > 0
```

Comment:
```suggestion
    if !isvalid(bytes)
        rawbytes = codeunits(bytes)
        n = count(≥(0x80), rawbytes)
```
I think this would fix the recursion problem. Since `!isvalid` implies that `n > 0` we don't need that check anymore.